demos/common/src/performance_metrics.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "samples/performance_metrics.hpp"
+
+#include <limits>
+
+// timeWindow defines the length of the timespan over which the 'current fps' value is calculated
+PerformanceMetrics::PerformanceMetrics(Duration timeWindow) : timeWindowSize(timeWindow) {}
+
+void PerformanceMetrics::update(TimePoint lastRequestStartTime,
+                                cv::Mat& frame,
+                                cv::Point position,
+                                double fontScale,
+                                cv::Scalar color,
+                                int thickness) {
+    TimePoint currentTime = Clock::now();
+
+    if (lastUpdateTime == TimePoint()) {
```

Comment:
I don't recommend using `TimePoint()` as a sentinel value, because that's the clock's epoch, which could occur as a normal time value returned by `Clock::now()` (which is prretty unlikely, but still). I would suggest to just use a boolean flag to track whether the first frame has been processed yet.